src/languages/pyret/PyretParser.ts

```diff
@@ -319,10 +319,10 @@ const nodeTypes = {
 
   // data Program
   "s-program": function(_pos: Loc, _provides: ASTNode, _provTy: any, imports: ASTNode[], body: Block) {
-    console.log("Provides in program:", _provides.length)
+    console.log("Provides in program:", _provides);
 
     let rootNodes = imports.concat(body.getExprs());
-    if (_provides.value != ""){
+    if ((_provides as $TSFixMe).value != ""){
```

Comment:
`ASTNodes` don't actually have a `value` field, so `ASTNode` is still the wrong type for `_provides`. But I couldn't figure out what the right type was quickly so I just left this as a `$TSFixMe`

Reply:
The actual type here can vary (maybe it shouldn't), but as of now I'm pretty sure it is either a `ProvideAll` (from the `s-provide-all` in the parse) or a literal that is really just a dummy placeholder (from `s-provide-none`). A better system would probably be to emit something other than an empty literal in the case of `s-provide-none` to represent no provides in the program.